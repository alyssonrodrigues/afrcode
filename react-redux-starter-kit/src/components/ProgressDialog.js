import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles'
import { connect } from 'react-redux'
import Dialog, { DialogContent } from 'material-ui/Dialog'
import CircularProgress from 'material-ui/Progress/CircularProgress'

import styles from './ProgressDialogStyles'

class ProgressDialog extends Component {
  render () {
    const { classes, operationProgress: { operationInProgress } } = this.props
    return (
      <Dialog PaperProps={{className: classes.paperClassName}}
        BackdropProps={{className: classes.backdropClassName}}
        open={operationInProgress}>
        <DialogContent>
          <CircularProgress />
        </DialogContent>
      </Dialog>
    )
  }
}

const materialUIEnhanced = withStyles(styles)(ProgressDialog)

const mapStateToProps = ({ operationProgress }) => ({ operationProgress })

export default connect(mapStateToProps)(materialUIEnhanced)
