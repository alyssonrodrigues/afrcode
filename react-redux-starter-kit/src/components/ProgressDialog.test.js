import React from 'react'
import { shallow } from 'enzyme'

import { ProgressDialog } from './ProgressDialog'

describe('ProgressDialog', () => {
  it('ProgressDialog render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <ProgressDialog classes={{}} operationProgress={{operationInProgress: true}} />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
