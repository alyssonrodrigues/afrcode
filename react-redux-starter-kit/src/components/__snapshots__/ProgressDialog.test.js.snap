// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressDialog ProgressDialog render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressDialog
    classes={Object {}}
    operationProgress={
        Object {
            "operationInProgress": true,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "BackdropProps": Object {
        "className": undefined,
      },
      "PaperProps": Object {
        "className": undefined,
      },
      "children": <WithStyles(DialogContent)>
        <WithStyles(CircularProgress) />
</WithStyles(DialogContent)>,
      "open": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CircularProgress) />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "BackdropProps": Object {
          "className": undefined,
        },
        "PaperProps": Object {
          "className": undefined,
        },
        "children": <WithStyles(DialogContent)>
          <WithStyles(CircularProgress) />
</WithStyles(DialogContent)>,
        "open": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CircularProgress) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
